app/src/main/java/gholzrib/arctouchchallenge/core/utils/PreferencesManager.java

```diff
@@ -0,0 +1,111 @@
+        SharedPreferences.Editor editor = preferences.edit();
+        editor.putInt(key, value);
+        editor.apply();
+    }
+
+    private static int getInt(Context context, String key, int defaultValue) {
+        SharedPreferences preferences = getDefaultPreferencesManager(context);
+        return preferences.getInt(key, defaultValue);
+    }
+
+    private static void setBoolean(Context context, String key, boolean value) {
+        SharedPreferences preferences = getDefaultPreferencesManager(context);
+        SharedPreferences.Editor editor = preferences.edit();
+        editor.putBoolean(key, value);
+        editor.apply();
+    }
+
+    private static boolean getBoolean(Context context, String key, boolean defaultValue) {
+        SharedPreferences preferences = getDefaultPreferencesManager(context);
+        return preferences.getBoolean(key, defaultValue);
+    }
+
+    private static void setString(Context context, String key, String value) {
+        SharedPreferences preferences = getDefaultPreferencesManager(context);
+        SharedPreferences.Editor editor = preferences.edit();
+        editor.putString(key, value);
+        editor.apply();
+    }
+
+    private static String getString(Context context, String key, String defaultValue) {
+        SharedPreferences preferences = getDefaultPreferencesManager(context);
+        return preferences.getString(key, defaultValue);
+    }
+
+    private static Boolean containsValue(Context context, String key){
+        SharedPreferences preferences = getDefaultPreferencesManager(context);
+        return preferences.contains(key);
+    }
+
+    public static void clearPreferences(Context context){
+        SharedPreferences preferences = getDefaultPreferencesManager(context);
+        SharedPreferences.Editor editor = preferences.edit();
+        editor.clear();
+        editor.apply();
+    }
+
+    public static void setTmdbConfiguration(Context context, TMDBConfiguration configuration){
+        String configJson = null;
+        if (configuration != null) {
+            Gson gson = new Gson();
+            configJson = gson.toJson(configuration);
+        }
+        setString(context, TMDB_CONFIGURATION_KEY, configJson);
+    }
+
+    public static TMDBConfiguration getTmdbConfiguration(Context context){
+        Gson gson = new Gson();
+        String json = getString(context, TMDB_CONFIGURATION_KEY, null);
+        if (json != null) {
+            try {
+                return gson.fromJson(json, TMDBConfiguration.class);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        return null;
+    }
+
+    public static Boolean containsTmdbConfiguration(Context context){
+        return containsValue(context, TMDB_CONFIGURATION_KEY);
+    }
+
+    public static void setTmdbConfigurationLastUpdateKey(Context context, String date) {
+        setString(context, TMDB_CONFIGURATION_LAST_UPDATE_KEY, date);
```

Comment:
This class was build this way to offer support to future data to be saved. In case another model, like the TMDBConfiguration, needed to be saved, it would be easy to create a specific method to receive this new model, while using the already created _setString_ to actually save it on the _SharedPreferences_.